### ISSUE TITLE
GlslangValidator generating invalid shaders for Intel

### ISSUE THREAD

**Issue by MarkY-LunarG** [KhronosGroup/glslang#798]
The issue appears to be that Nvidia specific byte code may be going into all vendor's shader outputs.  It's supposed to be in an extension specific packet, but AMD and Intel both seem to have had problems with it.

Intel provided the following command which shows the issue:

```
glslangValidator –V –H <any geom shader> | grep NV
```

This issue needs to be resolved before we can spin a new Vulkan SDK.

**Comment by johnkslang** [comment]
Isn't this a duplicate of the already well-understood issue that NV_EXTENSIONS is adding extra members to the interface?

I'm going to look at coding up a solution that requires both sides of the interface to declare the same #extension enabling this in order for it it work.

**Comment by pdaniell-nv** [comment]
What does "both sides of the interface" mean? I assume one side is the source using "#extension GL_vendor_extension : enable". What's the other side?

**Comment by johnkslang** [comment]
I mean, two stages sharing the interface.

I'm assuming that's what's hard about this, that it can't be done per compilation unit, and that's why Nvidia made the extra field always show up unconditionally.

So, I'm proposing that shaders of each involved stage have to declare the extension in order to get the interface to match.  It's still a bit tricky (hence clip and cull distance always present), but I think I can figure it out.

Is that plan okay?

**Comment by pdaniell-nv** [comment]
Yes, I think that's a good solution that glslang assumes that the shaders on both sides of the interface have correctly added "#extension GL_vendor : enable". Shaders that don't have this #extension declaration can't expect the interface to have the extension variables. In which case, glslang can just drop these extension variables on the interface if #extension isn't declared, which should resolve this problem for everyone.

Thanks!

**Comment by johnkslang** [comment]
@MarkY-LunarG I'm not finding your opening post to be true. Was it an older version of glslang?  I've tried a number of geometry shaders with getting NV stuff out.  Still looking at a fix though.

**Comment by MarkY-LunarG** [comment]
@scyganINTEL is the one who provided the initial repro steps.

**Comment by johnkslang** [comment]
Issue #790 more accurately characterizes the problem. Closing this one as a duplicate. If you can find a separate problem that is not captured by #790, post the specific code example that triggers it as a new issue.
